src/main/scala/akka/io/AsyncDnsResolver.scala

```diff
@@ -43,36 +48,58 @@ class AsyncDnsResolver(cache: SimpleDnsCache, config: Config) extends Actor with
   private def nextId(): Short = {
     do {
       requestId = (requestId + 2).toShort
-    } while (requests.contains(requestId))
+    } while (requests.exists(_._1 == requestId))
```

Comment:
Why worry at all about clashing from a paractical level? Nothing is guaranteed with UDP anyhow? Wdyt about just bumping a short and dealing with overflow ?